ci/check-changelog.sh

```diff
@@ -16,13 +16,46 @@ contains_changelog () {
     [[ 2 == $(git show -s --format=%B $1 | awk "$awk_script") ]]
 }
 
-for sha in $(git rev-list ${1:-origin/master}..); do
-    if contains_changelog $sha; then
-        echo "Commit $sha contains a changelog entry."
-        exit 0
+is_dependabot_pr() {
+    local key gpg_dir
+    if [ "1" = "$(git rev-list ${1:-origin/master}.. | wc -l)" ] && [ "dependabot[bot]" = "$(git show -s --format=%an)" ]; then
+        key=$(mktemp)
+        cat > $key <<DEPENDABOT
+-----BEGIN PGP PUBLIC KEY BLOCK-----
+
```

Comment:
Is this newline intentional? It probably doesn't matter but it looks odd to me.

Reply:
This is the verbatim output of `gpg --armor --export 4AEE18F83AFDEB23`.